torchvision/prototype/transforms/_geometry.py

```diff
@@ -309,61 +250,28 @@ def __init__(
         self.fill = fill
         self.padding_mode = padding_mode
 
-    def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
-        if isinstance(input, features.Image) or is_simple_tensor(input):
-            # PyTorch's pad supports only integers on fill. So we need to overwrite the colour
-            output = F.pad_image_tensor(input, params["padding"], fill=0, padding_mode="constant")
-
-            left, top, right, bottom = params["padding"]
-            fill = torch.tensor(params["fill"], dtype=input.dtype, device=input.device).to().view(-1, 1, 1)
-
-            if top > 0:
-                output[..., :top, :] = fill
-            if left > 0:
-                output[..., :, :left] = fill
-            if bottom > 0:
-                output[..., -bottom:, :] = fill
-            if right > 0:
-                output[..., :, -right:] = fill
-
-            if isinstance(input, features.Image):
-                output = features.Image.new_like(input, output)
-
-            return output
-        elif isinstance(input, PIL.Image.Image):
-            return F.pad_image_pil(
-                input,
-                params["padding"],
-                fill=tuple(int(v) if input.mode != "F" else v for v in params["fill"]),
-                padding_mode="constant",
-            )
-        elif isinstance(input, features.BoundingBox):
-            output = F.pad_bounding_box(input, params["padding"], format=input.format)
-
-            left, top, right, bottom = params["padding"]
-            height, width = input.image_size
-            height += top + bottom
-            width += left + right
-
-            return features.BoundingBox.new_like(input, output, image_size=(height, width))
-        else:
-            return input
+    def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
+        return F.pad(inpt, padding=self.padding, fill=self.fill, padding_mode=self.padding_mode)
 
 
 class RandomZoomOut(_RandomApplyTransform):
     def __init__(
-        self, fill: Union[float, Sequence[float]] = 0.0, side_range: Tuple[float, float] = (1.0, 4.0), p: float = 0.5
+        self,
+        fill: Union[int, float, Sequence[int], Sequence[float]] = 0,
+        side_range: Tuple[float, float] = (1.0, 4.0),
+        p: float = 0.5,
     ) -> None:
         super().__init__(p=p)
 
-        if fill is None:
-            fill = 0.0
+        _check_fill_arg(fill)
         self.fill = fill
 
         self.side_range = side_range
         if side_range[0] < 1.0 or side_range[0] > side_range[1]:
             raise ValueError(f"Invalid canvas side range provided {side_range}.")
 
+        self._pad_op = Pad(0, padding_mode="constant")
```

Comment:
I would probably avoid doing this. It puts an idiom of using transforms within transforms which is probably unnecessary now that we have a middle layer. This was added originally by Philip to avoid duplicating massive amount of code from Pad. There was also the idea of having different primitive non-random Transform classes which we are currently moving away from.